src/db.c

```diff
@@ -2555,3 +2754,40 @@ int bitfieldGetKeys(struct redisCommand *cmd, robj **argv, int argc, getKeysResu
     }
     return 1;
 }
+
+void dbGetStats(char *buf, size_t bufsize, redisDb *db, int full) {
+    size_t l;
+    char *orig_buf = buf;
+    size_t orig_bufsize = bufsize;
+    dictStats *mainHtStats = NULL;
+    dictStats *rehashHtStats = NULL;
+    dict *d;
+    dbIterator dbit;
+    dbIteratorInit(&dbit, db);
+    while ((d = dbIteratorNextDict(&dbit))) {
+        dictStats *stats = dictGetStatsHt(d, 0, full);
+        if (!mainHtStats) mainHtStats = stats;
```

Comment:
styling: i think i'd rather add a line break here, and maybe even curly brackets.